imap-types/src/core.rs

```diff
@@ -1399,31 +1399,31 @@ impl<'a> AsRef<str> for Charset<'a> {
     }
 }
 
-/// A `Vec` that always contains >= 1 elements.
-///
-/// Some messages in IMAP require a list of at least one element. We encoded these situations in a
-/// non-empty vector type to not produce invalid messages.
-///
-/// The `Debug` implementation equals `Vec` with an attached `+` at the end.
 #[cfg_attr(feature = "bounded-static", derive(ToStatic))]
 #[cfg_attr(feature = "serde", derive(Serialize, Deserialize))]
 #[derive(Clone, PartialEq, Eq, Hash)]
-pub struct NonEmptyVec<T>(pub(crate) Vec<T>);
+pub struct VecN<T, const N: usize>(pub(crate) Vec<T>);
```

Comment:
It looks like you couldn't decide whether you make this type public or not.

I'd would prefer to document `VecN` extensively and replace the doc of `Vec1` and `Vec2` with a short comment.

Reply:
Hehe, that's exactly what happend :-) Couldn't publicly alias a private type (of course). Good catch!

Reply:
doc(hidden) would be an option. But I like VecN and would prefer to keep it and document it properly.